test/e2e/apimachinery/flowcontrol.go

```diff
@@ -317,6 +335,58 @@ func createFlowSchema(f *framework.Framework, flowSchemaName string, matchingPre
 	}
 }
 
+func getFlowSchema(f *framework.Framework, name string) *flowcontrol.FlowSchema {
+	fs, err := f.ClientSet.FlowcontrolV1beta1().FlowSchemas().Get(context.TODO(), name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return fs
+}
+
+func getPriorityLevelConfig(f *framework.Framework, name string) *flowcontrol.PriorityLevelConfiguration {
+	pl, err := f.ClientSet.FlowcontrolV1beta1().PriorityLevelConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return pl
+}
+
+// waitForSteadyState repeatedly polls the API server to check if the newly
+// created flow schema and priority level have been seen by the APF controller
+// by checking: (1) the dangling priority level reference condition in the flow
+// schema status, and (2) metrics. The function times out after 30 seconds.
+func waitForSteadyState(f *framework.Framework, flowSchemaName string, priorityLevelName string) {
+	framework.ExpectNoError(wait.Poll(time.Second, 30*time.Second, func() (bool, error) {
+		fs := getFlowSchema(f, flowSchemaName)
+		if apihelpers.GetFlowSchemaConditionByType(fs, flowcontrol.FlowSchemaConditionDangling) != nil {
+			return false, nil
```

Comment:
and when we find the `Dangling` condition, do we also want to check that its `status` is set to `False`?

Reply:
That would be a nice add, yes.

Reply:
Done.

Reply:
When the dangling Condition exists with the value "true" this means the apiserver has processed the FlowSchema object and _not_ found the corresponding PriorityLevelConfiguration object.  In this case this func should `return false, nil` to cause a pause and retry.

When the dangling Condition exists with a value of "true", this func should `return true, nil` to cause successful completion of the `wait.Poll`.